examples/app-cpp/payload_app.cc

```diff
@@ -199,29 +247,9 @@ static void sequence_core(mythreadState_t *mythread)
     return;
```

Comment:
We may be leaving this function still holding onto cond_lock mutex. Not sure if it is intentional.